src/test/java/com/spring1/meliSocial/unitTest/service/PostServiceTest.java

```diff
@@ -1,4 +1,184 @@
 package com.spring1.meliSocial.unitTest.service;
 
+import com.spring1.meliSocial.dto.response.PostIndexDto;
+import com.spring1.meliSocial.dto.response.ResponsePostDto;
+import com.spring1.meliSocial.exception.BadRequestException;
+import com.spring1.meliSocial.exception.NotFoundException;
+import com.spring1.meliSocial.mapper.IMapper;
+import com.spring1.meliSocial.model.Post;
+import com.spring1.meliSocial.model.Product;
+import com.spring1.meliSocial.model.User;
+import com.spring1.meliSocial.repository.IPostRepository;
+import com.spring1.meliSocial.repository.IUserRepository;
+import com.spring1.meliSocial.service.impl.PostService;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.DisplayName;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import org.springframework.boot.test.context.SpringBootTest;
+
+import java.time.LocalDate;
+import java.util.*;
+
+import static org.junit.jupiter.api.Assertions.*;
+import static org.mockito.Mockito.when;
+
+@SpringBootTest
+@ExtendWith(MockitoExtension.class)
 public class PostServiceTest {
+
+    @Mock
+    IPostRepository postRepository;
+
+    @Mock
+    IUserRepository userRepository;
+
+    @Mock
+    IMapper customMapper;
+
+    @InjectMocks
+    PostService postService;
+
+    User seller1;
+    User user1;
+    User user9;
+    Post post1;
+    Post post2;
+    Post post17;
+    Post post18;
+    List<Post> postsEsperados;
+    ResponsePostDto postDto17;
+    ResponsePostDto postDto18;
+
+    @BeforeEach
+    public void setUp() {
+        seller1 = new User(1, "seller1", true, List.of(2), new ArrayList<>(), new ArrayList<>(), new HashSet<>());
+        user1 = new User(2, "user1", false, new ArrayList<>(), List.of(1), new ArrayList<>(), new HashSet<>());
+        post1 = new Post(1, 1, LocalDate.now(), null, 2, 100.0, false, 0.0);
+        post2 = new Post(2, 1, LocalDate.now(), null, 2, 250.0, false, 0.0);
+
+        //test de post x order
+        user9= new User(9, "Lucia Quezada", false, List.of(), List.of(8), List.of(), Set.of());
+        post17 = new Post(17, 8, LocalDate.parse("2025-01-01"), new Product(), 2, 20.99, true, 0.3);
```

Comment:
Y si usamos la fecha actual al momento de ejecutar los test? Asi nos ahorramos el tema este de tener que actualizar las fechas cuando dentro de x tiempo. Te quedaria algo asi donde necesites usar una fecha:
```suggestion
        post17 = new Post(17, 8, LocalDate.now(), new Product(), 2, 20.99, true, 0.3);
```
Que opinas?

Reply:
aaaaa tenes razón, m gusta